hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReaderSchemaHandler.java

```diff
@@ -199,7 +199,8 @@ private static String[] getMandatoryFieldsForMerging(HoodieTableConfig cfg, Type
       }
     }
 
-    if (cfg.getRecordMergeMode() == RecordMergeMode.EVENT_TIME_ORDERING) {
+    if (cfg.getRecordMergeMode() == null
+        || cfg.getRecordMergeMode() == RecordMergeMode.EVENT_TIME_ORDERING) {
```

Comment:
A better way is to return `EVENT_TIME_ORDERING` from `cfg.getRecordMergeMode()` for table version 6.

Reply:
Also, could you add a test that fails before this fix and passes after this fix?